packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/views/relatedelements/services/OutgoingTreeDescriptionProvider.java

```diff
@@ -118,18 +118,21 @@ private String getNodeLabel(VariableManager variableManager) {
         if (self instanceof EReference eReference) {
             result = this.labelService.getStyledLabel(eReference).toString();
             var optionalRootEObject = variableManager.get(VariableManager.SELF, EObject.class);
-            if (optionalRootEObject.isPresent()) {
-                result = this.getDisplayName(eReference, optionalRootEObject.get());
+            var optAdapterFactory = variableManager.get(IEditingContext.EDITING_CONTEXT, IEMFEditingContext.class)
+                    .map(IEMFEditingContext::getDomain)
+                    .map(AdapterFactoryEditingDomain::getAdapterFactory);
```

Comment:
It's small, but this pattern now gets repeated a lot. I wonder if we should add `IEMFEditingContext.getAdapterFactory`.